tests/unittests/BackendTest.cpp

```diff
@@ -165,6 +165,77 @@ TEST(RuntimeBundle, BundleSymbolInfo) {
   EXPECT_EQ(table.find("tensorview_reshape")->second.output, false);
 }
 
+// Test if the placeholders are allocated contiguously as
+// Input|InputOutput|Output.
+TEST(RuntimeBundle, ContiguousPlaceholder) {
+  ExecutionEngine EE;
+  auto &MD = EE.getModule();
+  auto *F = MD.createFunction("main");
+
+  unsigned depth = 16;
+  llvm::SmallVector<unsigned_t, 2> kernels = {5, 5};
+  llvm::SmallVector<unsigned_t, 4> pads = {0, 0, 0, 0};
+  llvm::SmallVector<unsigned_t, 2> steps = {1, 1};
+  unsigned width = 224;
+
+  auto *input = MD.createPlaceholder(ElemKind::Int8QTy, {1, width, width, 3},
+                                     0.4, 2, "Input", true);
+
+  // Calculate the size and allocate the output buffer.
+  std::array<size_t, 4> filterDim = {{depth, kernels[0], kernels[1], 3}};
+  auto *filter =
+      MD.createPlaceholder(ElemKind::Int8QTy, filterDim, 3.3, 4, "F", true);
+  auto *bias =
+      MD.createPlaceholder(ElemKind::Int32QTy, {depth}, 1.3, 5, "B", true);
+
+  // Calculate the size and allocate the output buffer.
+  auto outSz = calculateConvPoolOutputDims(width, width, kernels, steps, pads);
+  std::array<size_t, 4> outDims = {{1, outSz.first, outSz.second, 16}};
+  auto t = F->getParent()->uniqueType(glow::ElemKind::Int8QTy, outDims, 1.5, 6);
+
+  auto *conv =
+      F->createConv("conv", input, filter, bias, t, kernels, steps, pads, 1);
+
+  auto s = conv->getResult().getType()->size();
+  auto *fcFilter =
+      MD.createPlaceholder(ElemKind::Int8QTy, {s, 6}, 0.4, 2, "F", true);
+  auto *fcBias =
+      MD.createPlaceholder(ElemKind::Int32QTy, {6}, 0.4, 2, "B", true);
+  Node *O = F->createFullyConnected("fc1", conv, fcFilter, fcBias);
+  PlaceholderBindings bindings;
+  F->createSave("ret", O);
+  EE.compile(CompilationMode::Infer, F);
+  auto table = EE.getCompiledFunction().getRuntimeBundle().getSymbolTable();
+
+  std::vector<glow::runtime::RuntimeSymbolInfo> tableContainer;
+  // Only check placeholders.
+  for (auto v : table) {
+    if (v.second.input || v.second.output) {
+      tableContainer.push_back(v.second);
+    }
+  }
+  // Sort the placeholders by offset.
+  sort(tableContainer.begin(), tableContainer.end(),
+       [](const glow::runtime::RuntimeSymbolInfo &a,
+          const glow::runtime::RuntimeSymbolInfo &b) {
+         return (a.offset < b.offset);
+       });
+  // Check if the placeholders are allocated in the contiguous order
+  // Input|Inputoutput|Output.
+  int prev = -2;
```

Comment:
This is a test and the logic below is not very obvious. Please try to simplify it to make it obvious rather than fast or efficient.